### PR TITLE
Implement healthcheck endpoint

### LINKED ISSUE #3631: [FEAT] Implement Docker healthcheck
**Description**
Add a simple `/healthcheck` endpoint that simply returns 200 in case no error has been reported. In case of error, a different return code and a JSON returns otherwise.

**Motivation**
To easily spot and automatically restart unresponsive containers.

**Justification**
Such a feature would improve the monitoring of the chainlink node, letting your monitoring tool quick know and restart a given container. [[1]](https://docs.docker.com/engine/reference/builder/#healthcheck)
Most modern orchestration tools (ECS, K8S, etc) can read this health status and automatically restart the service. [[2]](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request) [[3]](https://docs.docker.com/config/daemon/prometheus/)

**Additional Information**
If the application is still responsive (working normally), a `docker ps` would indicate the healthy status:
``` bash
CONTAINER ID        IMAGE                            COMMAND                  CREATED             STATUS                   PORTS               NAMES
e2722770cc7f        smartcontract/chainlink:latest   "chainlink local node"   3 minutes ago       Up 3 minutes (healthy)                       chainlink
```

### PR DESCRIPTION
**Pull request by willianpaixao**
Fix #3631

Implement a `/healthcheck` endpoint and adds a Docker healthcheck check.

### CODE REVIEW

File: core/chainlink.Dockerfile [hunk 1]
```diff
@@ -66,12 +66,15 @@ RUN make chainlink-build
...
+RUN apt-get update && apt-get install -y ca-certificates curl
 
 WORKDIR /root
 
 COPY --from=1 /go/bin/chainlink /usr/local/bin/
 
 EXPOSE 6688
 ENTRYPOINT ["chainlink"]
+
+HEALTHCHECK CMD curl -f http://localhost:6688 || exit 1
```
Comment (jleeh) [inline core/chainlink.Dockerfile:78]: Would the healthcheck be able to run on a different port to the UI? There's an issue with using the 6688 port as it's only opened when the DB lock is grabbed and if you're running a standby node that waits indefinitely for the lock, it'd never be marked as healthy.

File: go.sum [hunk 2]
```diff
@@ -195,6 +195,7 @@ github.com/dgryski/go-farm v0.0.0-20190423205320-6a90982ecee2/go.mod h1:SqUrOPUn
 github.com/dgryski/go-sip13 v0.0.0-20181026042036-e10d5fee7954/go.mod h1:vAd38F8PWV+bWy6jNmig1y/TA+kYO4g3RSRF0IAv0no=
 github.com/dlclark/regexp2 v1.2.0/go.mod h1:2pZnwuY/m+8K6iRw6wQdMtk+rH5tNGR1i55kozfMjCc=
 github.com/docker/docker v1.4.2-0.20180625184442-8e610b2b55bf/go.mod h1:eEKB0N0r5NX/I1kEveEz05bcu8tLC/8azJZsviup8Sk=
+github.com/docker/go-healthcheck v0.1.0 h1:6ZrRr63F5LLsPwSlbZgjgoxNu+o1VlMIhCQWgbfrgU0=
```
Comment (jleeh) [inline go.sum:198]: Is this package being used in this PR, looks like it opens 5001 by default? Just looking at it, it'd be nice to use some of its features, for example adding a check that if the node is active, it's still connected to the ETH client.

**Review by jleeh — CHANGES_REQUESTED** [review]
Thanks for the PR Willian! Just left some comments that have been an historic issue with the current design, as horizontal scaling isn't supported.

File: go.sum (hunk 2 above)
Comment (willianpaixao) [reply go.sum:198]: Hey @jleeh, thanks for the review!
To be honest with you, I am not a Go developer, so I only implemented the basic use case described in the package's documentation. But I totally agree, this health check could be much smarter.

As for the opening of door 5001, I didn't see that. But unless you use the `EXPOSE` tag in the Dockerfile this port would never see the light of the day.

File: core/chainlink.Dockerfile (hunk 1 above)
Comment (willianpaixao) [reply core/chainlink.Dockerfile:78]: Yes, that would be possible, I just don't have the Go skills for that. :sweat_smile: 
Would you like to do that or point out a developer to continue the PR?

**Review by samsondav — CHANGES_REQUESTED** [review]
Realistically this is never going to be approved as long as it pulls in a third party lib that opens ports on its own (it's a security concern).

Happy to have a health check though, if you want to implement through our existing routing.

**Comment by willianpaixao** [comment]
Hej @samsondav, thanks for giving attention to this forgotten PR!

This port is never exposed to outside the container, only the curl running inside can see it.

**Comment by samsondav** [comment]
Not everybody runs chainlink in docker. We don't want to be pulling in any additional attack surfaces.

If you want to implement a health check you are more than welcome to do it as an addition using our existing controller patterns. Otherwise we should close this PR.

**Comment by willianpaixao** [comment]
Fair point. I'll close the isue.

**Pull request closed without merging**